Description: Prevent 32-bit overflows in tests due to untyped ints
Author: Daniel Swarbrick <dswarbrick@debian.org>
Bug: https://github.com/vultr/govultr/issues/147
Last-Update: 2022-10-22
--- a/backup_test.go
+++ b/backup_test.go
@@ -20,7 +20,7 @@ func TestBackupServiceHandler_List(t *te
 				"id": "543d34149403a",
 				"date_created": "2014-10-14 12:40:40",
 				"description": "Automatic server backup",
-				"size": 42949672960,
+				"size": 2147483647,
 				"status": "complete"
 				}
 			],
@@ -47,7 +47,7 @@ func TestBackupServiceHandler_List(t *te
 			ID:          "543d34149403a",
 			DateCreated: "2014-10-14 12:40:40",
 			Description: "Automatic server backup",
-			Size:        42949672960,
+			Size:        2147483647,
 			Status:      "complete",
 		},
 	}
@@ -124,7 +124,7 @@ func TestBackupServiceHandler_Get(t *tes
 				"id": "543d34149403a",
 				"date_created": "2014-10-14 12:40:40",
 				"description": "Automatic server backup",
-				"size": 42949672960,
+				"size": 2147483647,
 				"status": "complete"
 			}
 		}
@@ -142,7 +142,7 @@ func TestBackupServiceHandler_Get(t *tes
 		ID:          "543d34149403a",
 		DateCreated: "2014-10-14 12:40:40",
 		Description: "Automatic server backup",
-		Size:        42949672960,
+		Size:        2147483647,
 		Status:      "complete",
 	}
 
--- a/bare_metal_server_test.go
+++ b/bare_metal_server_test.go
@@ -30,7 +30,7 @@ func TestBareMetalServerServiceHandler_G
 				"v6_network": "2001:DB8:9000::",
 				"v6_main_ip": "2001:DB8:9000::100",
 				"v6_network_size": 64,
-				"mac_address": 2199756823533,
+				"mac_address": 2147483647,
 				"label": "my label",
 				"tags": ["my tag"],
 				"os_id": 127,
@@ -62,7 +62,7 @@ func TestBareMetalServerServiceHandler_G
 		V6Network:     "2001:DB8:9000::",
 		V6MainIP:      "2001:DB8:9000::100",
 		V6NetworkSize: 64,
-		MacAddress:    2199756823533,
+		MacAddress:    2147483647,
 		Label:         "my label",
 		Tags:          []string{"my tag"},
 		OsID:          127,
@@ -270,7 +270,7 @@ func TestBareMetalServerServiceHandler_L
 					"v6_network": "2001:DB8:9000::",
 					"v6_main_ip": "2001:DB8:9000::100",
 					"v6_network_size": 64,
-					"mac_address": 2199756823533,
+					"mac_address": 2147483647,
 					"label": "my label",
 					"tags": ["my tag"],
 					"os_id": 127,
@@ -304,7 +304,7 @@ func TestBareMetalServerServiceHandler_L
 			V6Network:     "2001:DB8:9000::",
 			V6MainIP:      "2001:DB8:9000::100",
 			V6NetworkSize: 64,
-			MacAddress:    2199756823533,
+			MacAddress:    2147483647,
 			Label:         "my label",
 			Tags:          []string{"my tag"},
 			OsID:          127,
--- a/snapshot_test.go
+++ b/snapshot_test.go
@@ -12,7 +12,7 @@ func TestSnapshotServiceHandler_Create(t
 	defer teardown()
 
 	mux.HandleFunc("/v2/snapshots", func(writer http.ResponseWriter, request *http.Request) {
-		response := `{"snapshot":{"id": "5359435d28b9a","date_created": "2014-04-18 12:40:40","description": "Test snapshot","size": 42949672960,"compressed_size": 1078864689,"status": "complete","os_id": 127,"app_id": 0}}`
+		response := `{"snapshot":{"id": "5359435d28b9a","date_created": "2014-04-18 12:40:40","description": "Test snapshot","size": 2147483647,"compressed_size": 1078864689,"status": "complete","os_id": 127,"app_id": 0}}`
 		fmt.Fprint(writer, response)
 	})
 
@@ -30,7 +30,7 @@ func TestSnapshotServiceHandler_Create(t
 		ID:             "5359435d28b9a",
 		DateCreated:    "2014-04-18 12:40:40",
 		Description:    "Test snapshot",
-		Size:           42949672960,
+		Size:           2147483647,
 		CompressedSize: 1078864689,
 		Status:         "complete",
 		OsID:           127,
@@ -47,7 +47,7 @@ func TestSnapshotServiceHandler_CreateFr
 	defer teardown()
 
 	mux.HandleFunc("/v2/snapshots/create-from-url", func(writer http.ResponseWriter, request *http.Request) {
-		response := `{"snapshot":{"id": "5359435d28b9a","date_created": "2014-04-18 12:40:40","description": "Test snapshot","size": 42949672960,"compressed_size" : 1078864689,"status": "complete","os_id": 127,"app_id": 0}}`
+		response := `{"snapshot":{"id": "5359435d28b9a","date_created": "2014-04-18 12:40:40","description": "Test snapshot","size": 2147483647,"compressed_size" : 1078864689,"status": "complete","os_id": 127,"app_id": 0}}`
 		fmt.Fprint(writer, response)
 	})
 	snap := SnapshotURLReq{URL: "http://vultr.com"}
@@ -60,7 +60,7 @@ func TestSnapshotServiceHandler_CreateFr
 		ID:             "5359435d28b9a",
 		DateCreated:    "2014-04-18 12:40:40",
 		Description:    "Test snapshot",
-		Size:           42949672960,
+		Size:           2147483647,
 		CompressedSize: 1078864689,
 		Status:         "complete",
 		OsID:           127,
@@ -77,7 +77,7 @@ func TestSnapshotServiceHandler_Get(t *t
 	defer teardown()
 
 	mux.HandleFunc("/v2/snapshots/5359435d28b9a", func(writer http.ResponseWriter, request *http.Request) {
-		response := `{"snapshot":{"id": "5359435d28b9a","date_created": "2014-04-18 12:40:40","description": "Test snapshot","size": 42949672960,"compressed_size": 1078864689,"status": "complete","os_id": 127,"app_id": 0}}`
+		response := `{"snapshot":{"id": "5359435d28b9a","date_created": "2014-04-18 12:40:40","description": "Test snapshot","size": 2147483647,"compressed_size": 1078864689,"status": "complete","os_id": 127,"app_id": 0}}`
 		fmt.Fprint(writer, response)
 	})
 
@@ -90,7 +90,7 @@ func TestSnapshotServiceHandler_Get(t *t
 		ID:             "5359435d28b9a",
 		DateCreated:    "2014-04-18 12:40:40",
 		Description:    "Test snapshot",
-		Size:           42949672960,
+		Size:           2147483647,
 		CompressedSize: 1078864689,
 		Status:         "complete",
 		OsID:           127,
@@ -122,7 +122,7 @@ func TestSnapshotServiceHandler_List(t *
 	defer teardown()
 
 	mux.HandleFunc("/v2/snapshots", func(writer http.ResponseWriter, request *http.Request) {
-		response := `{"snapshots": [{"id": "885ee0f4f263c","date_created": "2014-04-18 12:40:40","description": "Test snapshot","size": 42949672960,"compressed_size": 1078864689,"status": "complete","os_id": 127,"app_id": 0}],"meta": {"total": 4,"links": {"next": "","prev": ""}}}`
+		response := `{"snapshots": [{"id": "885ee0f4f263c","date_created": "2014-04-18 12:40:40","description": "Test snapshot","size": 2147483647,"compressed_size": 1078864689,"status": "complete","os_id": 127,"app_id": 0}],"meta": {"total": 4,"links": {"next": "","prev": ""}}}`
 		fmt.Fprint(writer, response)
 	})
 
@@ -136,7 +136,7 @@ func TestSnapshotServiceHandler_List(t *
 			ID:             "885ee0f4f263c",
 			DateCreated:    "2014-04-18 12:40:40",
 			Description:    "Test snapshot",
-			Size:           42949672960,
+			Size:           2147483647,
 			CompressedSize: 1078864689,
 			Status:         "complete",
 			OsID:           127,
